python/social_bot/envs/pr2.py

```diff
@@ -58,128 +81,288 @@ def __init__(self,
+        self._resized_image_size = (128, 128) #(84, 84)
+        self._use_internal_states_only = use_internal_states_only
         self._cum_reward = 0.0
+        self._motion_loss = motion_loss
+
+        # a hack to walk around gripper not open initially, might not need now.
+        self._gripper_reward_dir = 1
+        self._gripper_upper_limit = 0.07
+        self._gripper_lower_limit = 0.01
 
-        img = self._get_observation()
-        self.observation_space = gym.spaces.Box(
-            low=0, high=1, shape=img.shape, dtype=np.uint8)
+        # whether to move head cameras and gripper when episode starts.
+        self._adjust_position_at_start = False
+
+        obs = self._get_observation()
+        self._prev_dist = self._get_finger_tip_distance()
+        self._prev_gripper_pos = self._get_gripper_pos()
+
+        if self._use_internal_states_only:
+            self.observation_space = gym.spaces.Box(
+                low=-np.inf, high=np.inf, shape=obs.shape, dtype=np.float32)
+        else:
+            self.observation_space = gym.spaces.Box(
+                low=0, high=256.0, shape=obs.shape, dtype=np.float32)
         self.action_space = gym.spaces.Box(
-            low=-5, high=5, shape=[len(self._r_arm_joints)], dtype=np.float32)
+            low=-1, high=1, shape=[len(self._r_arm_joints)], dtype=np.float32)
 
     def reset(self):
         self._world.reset()
-        # move camera to focus on ground. it is hacky
-        joint_state = gazebo.JointState(1)
-        joint_state.set_positions([1.4])
-        joint_state.set_velocities([0.0])
-        self._collision_cnt = 0
-        self._agent.set_joint_state("pr2::pr2::head_tilt_joint", joint_state)
-        self._world.step(200)
+        self._move_goal()
+        self._grip = False
 
+        if self._adjust_position_at_start:
+            # move camera to focus on ground. it is hacky
+            joint_state = gazebo.JointState(1)
+            joint_state.set_positions([1.4])
+            joint_state.set_velocities([0.0])
+            self._agent.set_joint_state("pr2::pr2::head_tilt_joint", joint_state)
+
+            self._agent.take_action(dict({"pr2::pr2::r_gripper_joint":5}))
+
+        self._world.step(200)
         self._goal = self._world.get_agent(self._goal_name)
-        self._move_goal()
+
+        obs = self._get_observation()
+
+        self._table_height = self._goal_pose[0][2]
+        #logger.debug("table height:" + str(self._table_height))
         self._steps_in_this_episode = 0
         self._cum_reward = 0.0
+
         self._prev_dist = self._get_finger_tip_distance()
-        return self._get_observation()
+        self._prev_gripper_pos = self._get_gripper_pos()
+        self._gripper_reward_dir = 1
+        return obs
 
     def _move_goal(self):
-        loc = (0.6 + 0.3 * (random.random() - 1), 0.4 * (random.random() - 1),
-               0)
+        loc = (0.76 + 0.14 * (random.random() - 1),  0.1 * (random.random() - 1.0),
```

Comment:
The PR2 robot is at (0, 0), and this base is pretty big. The table is (0.9, 0), since we only control the right arm, so we put the goal object on the right side (0.76 + 0.14 (random - 0.5)).

Reply:
Cannot understand your response